Tasks/Common/azure-arm-rest/azure-arm-mysql.ts

```diff
@@ -0,0 +1,320 @@
+            tl.debug("Response for add firewall rule " +statusCode);
+            if (statusCode != 200 && statusCode != 201 && statusCode != 202) {
+                // Generate Error
+                deferred.reject(new azureServiceClient.ApiResult(azureServiceClient.ToError(response)));
+            }
+            else if(statusCode === 202){
+                this._recursiveGetCall(resourceGroupName, serverName, firewallRuleName, 3).then((response) => {
+                    deferred.resolve(new azureServiceClient.ApiResult(null, response));
+                },(error) => {
+                    deferred.reject(error);
+                });
+            }
+            else {
+                // Generate Response
+                deferred.resolve(new azureServiceClient.ApiResult(null, response.body));
+            }
+            return deferred.promise;
+        }).then((apiResult: azureServiceClient.ApiResult) => callback(apiResult.error, apiResult.result),
+            (error) => callback(error));
+    }
+
+    /**
+     * Delete firewall rule of mysql server
+     * @param resourceGroupName     resource group name of mysql server 
+     * @param serverName            mysql server name
+     * @param firewallRuleName      firewall rule name to be deleted 
+     * @param callback              response callback 
+     */
+    public delete(resourceGroupName: string, serverName: string, firewallRuleName: string, callback?:  azureServiceClient.ApiCallback) {
+        var client = this.client;
+        if (!callback) {
+            throw new Error(tl.loc("CallbackCannotBeNull"));
+        }
+
+        // Validate
+        try {
+            this.client.isValidResourceGroupName(resourceGroupName);
+            if(!this.isNameValid(serverName)){
+                throw new Error(tl.loc("MysqlServerNameCannotBeEmpty"));
+            }
+            if(!this.isNameValid(firewallRuleName)){
+                throw new Error(tl.loc("FirewallRuleNameCannotBeNull"));
+            }
+        }
+        catch (error) {
+            return callback(error);
+        }
+
+        var httpRequest = new webClient.WebRequest();
+        httpRequest.method = 'DELETE';
+        httpRequest.headers = {};
+        httpRequest.uri = this.client.getRequestUri(
+            '//subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DBforMySQL/servers/{serverName}/firewallRules/{firewallRuleName}',
+            {
+                '{resourceGroupName}': resourceGroupName,
+                '{serverName}': serverName,
+                '{firewallRuleName}': firewallRuleName
+            });
+
+        this.client.beginRequest(httpRequest).then(async (response) => {
+            var deferred = Q.defer<azureServiceClient.ApiResult>();
+            var statusCode = response.statusCode;
+            tl.debug("Response for delete firewall rule " +statusCode);
+            if (statusCode != 200 && statusCode != 202 && statusCode !=204) {
```

Comment:
202 is required? if so, we must poll then

Reply:
Delete is last operation in our task so there is no need to poll for them. But in case of add rule we will able to run our script only it has successfully added.